src/analyze.ts

```diff
@@ -135,15 +135,14 @@ export function findExports (code: string): ESMExport[] {
     return []
   }
 
-  return exports.filter((exp, index, exports) => {
+  return exports
     // Filter false positive export matches
-    if (exportLocations && !_isExportStatement(exportLocations, exp)) {
-      return false
-    }
+    .filter(exp => !exportLocations || _isExportStatement(exportLocations, exp))
     // Prevent multiple exports of same function, only keep latest iteration of signatures
-    const nextExport = exports[index + 1]
```

Comment:
We were dedupe with the one that filtered out by AST. Changed to two passes of `.filter` to ensure `exports` is the real exports we want for dedupe.